src/components/elements/upload/UploaderBase.tsx

```diff
@@ -88,157 +47,108 @@ const getFileTypesFromAccept = (accept: Accept) => {
-            )}
-            <Typography color='inherit'>{file?.name}</Typography>
-            <Tooltip title='Clear uploaded image'>
-              <Link
-                component='button'
-                underline='none'
-                onClick={(e) => {
-                  e.stopPropagation();
-                  clearFile();
-                }}
-                variant='body2'
-                color='GrayText'
-                sx={{
-                  display: 'inline-flex',
-                  alignItems: 'center',
-                  mt: 1,
-                }}
-              >
-                <HighlightOffIcon fontSize='inherit' />
-                &nbsp;Clear
-              </Link>
-            </Tooltip>
-          </>
-        )}
-        {loading && (
-          <>
-            <Typography variant='subtitle1' color='inherit' gutterBottom>
-              Uploading
-            </Typography>
-            <LinearProgress variant='indeterminate' />
-          </>
-        )}
-        {error && (
-          <Typography variant='subtitle1' color='error' sx={{ mt: 2 }}>
-            {error}
-          </Typography>
-        )}
-      </Grid>
-    </Grid>
-  </Box>
-);
+export type ChildrenArgs = {
+  id: string;
+  currentFiles?: File[];
+  existingFiles?: FileFieldsFragment[];
+  loading?: boolean;
+  errors?: string[] | undefined;
+  dragging?: boolean;
+  removeFile?: (file: File) => void;
+  selectFiles?: VoidFunction;
+  rootProps?: DropzoneRootProps;
+  inputProps?: DropzoneInputProps;
+  accept?: Accept;
+  maxSize: number;
+  multiple?: boolean;
+};
+
+export type UploaderProps = {
+  id: string;
+  files?: (string | FileFieldsFragment)[];
+  onChange?: (files: (string | FileFieldsFragment)[]) => void;
+  onUpload?: (uploads: DirectUpload[], files: File[]) => any | Promise<any>;
+  accept?: Accept;
+  image?: boolean;
+  maxSize?: number;
+  multiple?: boolean;
+  children?: React.ReactNode | ((args: ChildrenArgs) => React.ReactElement);
+};
 
 const Uploader: React.FC<UploaderProps> = ({
   id,
+  files,
+  onChange,
   onUpload,
-  onClear = () => {},
-  file: fileProp,
-  image: isImage = false,
   accept: acceptProp,
-  maxSize = DEFUALT_MAX_BYTES,
+  image: isImage = false,
+  maxSize = DEFAULT_MAX_BYTES,
+  multiple = false,
 }) => {
-  const [currentFile, setCurrentFile] = useState<File>();
-  const [currentUpload, setCurrentUpload] = useState<DirectUpload>();
+  // The uploader accepts a `files` argument which can contain either:
+  // - a STRING which points at a blob ID of a file that has been uploaded within this session, or
+  // - a FileFieldsFragment record which points at a file record in our database, uploaded during a previous session.
+  // `existingFiles` filters the list to only those files that were uploaded some previous time, so we can render them.
+  // The files uploaded during this session, we render this component's internal state, `currentFiles`.
+  const existingFiles = useMemo(
```

Comment:
Ok, so, save without submitting is the major functionality that is still not working. I want to confirm the current (messy!!) approach before continuing to work on that. But here I'll try to briefly summarize the current behavior:

- Right now if you upload a file then "save without submitting," then reopen the assessment, you don't see the file. However, if you then save, you do see the file -- it successfully saved!
- What's going on under the hood?
  - When you first upload a file, the CDE `value` is a string -- the blob ID. `onChange` callback is called, and the CDE value in memory gets updated to point at that string -- but this component ignores that (filter on line 94), because it's rendering the current upload using the File object coming from dropzone.
  - When you actually _submit_ the assessment, it gets saved as a File record. And then the next time you fetch the assessment, the value is returned as a FileFieldsFragment type.
  - However, when you just _save_ the assessment, the File record doesn't get saved. The value coming back when you reopen is a string, the blob ID, which is filtered out here on line 94. However, we don't have a `currentFile` or `currentUpload` for it either, so it's just not rendered at all.
  - But it's still present on the CDE's `value` here in the frontend in memory, so when you _do_ eventually _submit_ the assessment, it is saved and you can view it when you reopen the assessment. 😵 

Maybe this indicates that something about the approach here is rotten to the core?